packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts

```diff
@@ -0,0 +1,221 @@
+import { z } from 'zod';
+import { invariant } from 'outvariant';
+import { isErrorWithMessage, ProcessStatus } from '@ballerine/common';
+
+import { logger } from '../../../logger';
+import { ApiPlugin } from '../api-plugin';
+import { TContext } from '../../../utils/types';
+import { validateEnv } from '../shared/validate-env';
+import { getTransformer } from '../../../workflow-runner-utils';
+import { IApiPluginParams, PluginPayloadProperty } from '../types';
+import { getPayloadPropertiesValue } from '../shared/get-payload-properties-value';
+import { handleJmespathTransformers } from '../shared/handle-jmespath-transformers';
+
+const BankAccountVerificationPluginPayloadSchema = z.object({
+  clientId: z.string().min(1),
+  vendor: z.enum(['experian']),
+  address: z.object({
+    streetNumber: z.string().min(1),
+    street: z.string().min(1),
+    city: z.string().min(1),
+    postcode: z.string().min(1),
+  }),
+  bankAccountDetails: z.object({
+    holder: z.union([
+      z.union([
+        z.object({
+          bankAccountName: z.string().min(1),
+          companyRegistrationNumber: z.string().min(1),
+        }),
+        z.object({
+          bankAccountName: z.string().min(1),
+          registeredCharityNumber: z.string().min(1),
+        }),
+      ]),
+      z.object({
+        firstName: z.string().min(1),
+        middleName: z.string().optional(),
+        lastName: z.string().min(1),
+      }),
+    ]),
+    sortCode: z.string().min(1),
+    bankAccountNumber: z.string().min(1),
+  }),
+});
+
+type TBankAccountVerificationPluginPayload = {
+  clientId: PluginPayloadProperty;
+  vendor: PluginPayloadProperty;
+  data: {
+    address: {
+      streetNumber: PluginPayloadProperty;
+      street: PluginPayloadProperty;
+      city: PluginPayloadProperty;
+      postcode: PluginPayloadProperty;
+    };
+    bankAccountDetails: {
+      sortCode: PluginPayloadProperty;
+      bankAccountNumber: PluginPayloadProperty;
+    } & (
+      | {
+          holder: {
+            firstName: PluginPayloadProperty;
+            middleName: PluginPayloadProperty<string | undefined>;
+            lastName: PluginPayloadProperty;
+          };
+        }
+      | ({ bankAccountName: PluginPayloadProperty } & (
+          | { companyRegistrationNumber: PluginPayloadProperty }
+          | { registeredCharityNumber: PluginPayloadProperty }
+        ))
+    );
+  };
+};
+
+export class BankAccountVerificationPlugin extends ApiPlugin {
+  public static pluginType = 'http';
+  public payload: TBankAccountVerificationPluginPayload;
+
+  private pluginName = 'Bank Account Verification Plugin';
+
+  constructor({
+    payload,
+    ...pluginParams
+  }: IApiPluginParams & { payload: BankAccountVerificationPlugin['payload'] }) {
+    super({
```

Comment:
There are a lot of issues with the block - as a newcomer ill be very confused in that point (im aware its probably something you saw in other places, but lets try cleanning up a bit)

Something the pops out right away is why its not a part of the parent constructor.
It seems like something weve done with transfomers to be quick but shouldve been integral  part of the plugin system